wikimedia/wikipedia-ios
# [YiR] Share slide

**Phabricator:** https://phabricator.wikimedia.org/T376300

### Notes
* This PR adds the ability to share a slide using a system UIActivityViewController 
* For that, I'm creating a separate View derived from the slide, creating a snapshot and saving it as a UIImage
* After discussing with Design, we do not have a "preview" step and we're always setting the shared image size to an iPhone-sized image, even from iPad
* I'm adding the bottom card with the hashtag, even though it was nice to have 

### Test Steps
1. Set device to either Italy, Mexico
2. Set primary app language to a supported lang - en, it, es 
3. Tap through 5+ articles 
4. Go to the developer settings menu, enable the feature flag
5. Background/foreground the app to populate the report
6. Tap profile, you should see the entry point
7. On a slide, click share. Share to a message, slack
8. Verify UI and text
9. Test from iPad
